py-*: py39 support for 15 packages

erikbs opened the pull request:
#### Description

Adds py39 support for these Python packages:
* altgraph
* beaker
* cppy
* cycler
* gnureadline
* kiwisolver
* macholib
* mako
* matplotlib
* modulegraph
* olefile
* Pillow
* py2app
* tornado
* tornado5

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.9.5 13F1911
Xcode 6.2 6C131e

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

erikbs commented:
The logs reveal few details, but it looks like the tests failed on py-tornado. Apparently I forgot to run `port lint` on that one, because when I did that now I got the following error:

```Error: conflicts references unknown port: py39-tornado5```

I fixed this by pushing an update to py-tornado5 as well. Now both `py39-tornado` and `py39-tornado5`pass `port lint`.

ryandesign commented:
Thanks.

Note that you [missed some dependencies](https://build.macports.org/builders/jobs-mirror/builds/209044/steps/mirror/logs/stdio) that come in through variants:

```
--->  Fetching distfiles for py39-matplotlib
Error: No such port: py39-pyqt4
Error: No such port: py39-pyqt5
Error: No such port: py39-pyside
```

erikbs commented:
Thanks for letting me know, I did not notice that. Is there a way to list dependencies of all variants? I manually called `info py38-*` and updated all dependencies I could find, then used `install py39-*` to catch the rest (or so I thought). Qt5 failed to install on my system (10.9.5) the last time I tried, so I doubt that I will be able to use this method to correct the missing dependencies for py39-pyqt5.

To make things easier for myself I wrote a quick and dirty Python 3 script (to be run in the root directory of the repository, i.e. `macports-ports`) that finds all Python packages that support py38, but not py39. With two additional lines of code (#13 and #14) it can update those packages with py39 support as well. Can it be used or is this considered unsafe?

```
import re
from glob import glob
L = []
for pkg in glob('python/py-*'):
	try:
		with open('%s/Portfile' % pkg) as f: r=f.read()
	except: continue
	m = re.findall('python.versions[\d ]+', r)
	if len(m):
		m = m[0].split(' ')
		if '38' in m and not '39' in m:
			L.append(pkg.split('/')[-1])
			r = re.sub('(python.versions[\d ]+)', r'\1 39', r)
			with open('%s/Portfile' % pkg, 'w') as f: f.write(r)

print('Found %u packages that supported py38, but not py39' % len(L))
with open('pkglst.txt', 'w') as f: f.write('\n'.join(L))
```

ryandesign commented:
> Is there a way to list dependencies of all variants?

No.

> Can it be used or is this considered unsafe?

You can use this to help you identify ports that you want to update, but please verify that the ports install on your system before submitting a pull request and also manually look at the portfiles and verify that you've updated any dependencies first.

We probably don't want to try to update all ports at once. Let's do it in small testable batches.

erikbs commented:
> Let's do it in small testable batches.
Yes, that is most likely the best. I assumed that py39 was fully compatible with py38, but then I tested `py39-shiboken` and it failed because py39 silently removed some deprecated stuff.

I found out that I could get a list of most of the dependencies by simply requesting all variants when calling `port info` (even when they are incompatible, e.g. `port info py39-gobject +quartz +x11`). I created a new pull request (#8865) that hopefully solves the last dependencies for `py-matplotlib` without introducing new dependency issues.
